### PR TITLE
#161967009 Add descriptive validation error messages

### PR DESCRIPTION
**Pull request by mashafrancis**
#### What does this PR do?
This PR adds descriptive error validation messages for the register endpoint.
#### Description of Task to be completed?

- Write unit tests for the validation error messages
- Implement the error messages for the register endpoint

#### How should this be manually tested?
On the terminal, run the following commands:
```
$ git clone https://github.com/andela/ah-jumanji.git
$ git checkout ft-descriptive-validaton-errors-161967009
$ cd ah-jumanji
$ python3 -m venv venv
$ source venv/bin/activate
(venv) $ pip install -r requirements.txt
(venv) $ touch .env
(venv) $ echo "DATABASE_URL=postgres://username:password@localhost/db_name" >> .env
(venv) $ echo "DEBUG=on" >> .env
(venv) $ python manage.py migrate
(venv) $ python manage.py runserver
```
Using [Postman](https://www.getpostman.com) navigate to:
```
{your-server-root}:8000/api/users/register
```
with a payload of this format

```
{
    "username": "your-username",
    "email": "your-email",
    "password": "your-password"
}
```
for example: 
```
{
    "username": "jake",
    "email": "jake@jake.com",
    "password": "Jakejake1#"
}
```
#### What are the relevant pivotal tracker stories?
[#161967009](https://www.pivotaltracker.com/story/show/161967009)
#### Screenshots (if appropriate)

1. Register with missing fields
screenshot 2018-12-03 at 23 44 59

2. Register with invalid fields
screenshot 2018-12-05 at 16 54 48

3. Register with valid fields
screenshot 2018-12-05 at 16 55 40

4. Duplicate registration with the same fields
screenshot 2018-12-05 at 16 55 59

#### Checklist:

- [x] My code follows the style guidelines of this project
- [x] At least 2 people have reviewed my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] My PR has one commit.

### CODE REVIEW

**Review by EmmanuelBeja — COMMENTED** [review]
Nice work. you should be good to go after fixing the Hound issues.

**Review by ghost — COMMENTED** [review]
This is good, pending the `Hound CI` issues

**Comment by GransonO** [comment]
Looking great, however, the hound keeps barking on issues that can be captured by following the **`pep8`** guidelines.  If possible, check the linting in the local dev environment before pushing. Otherwise great work

**Comment by mashafrancis** [comment]
Thanks. @GransonO I have pep8 locally, however, I have just realised my pep8 is configured at 120 characters length, which is slightly different from flake8 (79 characters). I'm working on it now to comply with the guidelines and to keep our hound quiet.

**Comment by fabzer0** [comment]
Great work

**Review by gitau-kinuthia — APPROVED** [review]
The PR has corrected most of the original styling issues identified by Hound CI and implements the requested functionality. The PR looks good

**Comment by EmmanuelBeja** [comment]
Nice work fixing the issues

**Review by fabzer0 — APPROVED** [review]

File: authors/apps/authentication/tests/test_registration.py [hunk 1]
```diff
@@ -22,38 +26,160 @@ class TestRegistration:
...
-        response2 = test_client.post(reverse('register'), self.user, format='json')
+        """Test API cannot add an existing user"""
+        test_client.post(reverse('register'), self.user, format='json')
+        res = test_client.post(reverse('register'), self.user, format='json')
+
+        assert res.status_code == 400
+        assert 'errors' in res.data
+
+        assert res.data['errors']['email'][0] == \
+               "Email already exists. Sign in instead or try another."
```
[the remark this answers is not in the document]
Comment (GransonO) [reply authors/apps/authentication/tests/test_registration.py]: I see the Hound is still complaining. What did you agree with `TTL` to implement as a solution to this?

**Review by GransonO — APPROVED** [review]
From the most recent changes I have observed that the issues have reduced considerably. The PR is up to standards as expected. Great work **`LGTM`**

**Review by ghost — APPROVED** [review]
`LGTM`

**Review by EmmanuelBeja — APPROVED** [review]

**Pull request merged**